lib/semantics/symbol.cc

```diff
@@ -179,15 +179,19 @@ Symbol *GenericDetails::CheckSpecific() {
 
 void GenericDetails::CopyFrom(const GenericDetails &from) {
   if (from.specific_) {
-    CHECK(!specific_);
+    CHECK(!specific_ || specific_ == from.specific_);
```

Comment:
It looks like a bit brutal to me to die if `specific_ != from.specific_`. Are we sure this case was handled before ?
The following code, where `g` appears as two different specific proc, crashes f18 for me:
```
module m9a
  interface g
    module procedure s1
    module procedure g
  end interface
contains
  subroutine g()
  end
  subroutine s1(x)
    integer :: x
  end
end module
module m9b
  interface g
    module procedure s2
    module procedure g
  end interface
contains
  subroutine s2(x)
    real :: x
  end
  subroutine g(i)
    integer :: i
  end
end module
! Merge use-associated generics that have the same symbol (s1)
subroutine s9
  use m9a
  use m9b
end
```

Reply:
Yes, you're right. I'll change it to check for these cases before calling CopyFrom.

Reply:
Now we report an error in this case.